File: Backend3/src/main/java/edu/cit/backend3/models/Warning.java
package edu.cit.backend3.models;

import jakarta.persistence.*;
import lombok.Data;
import com.fasterxml.jackson.annotation.JsonIdentityInfo;
import com.fasterxml.jackson.annotation.ObjectIdGenerators;

import java.time.LocalDateTime;

@Entity
@Data
@Table(name = "warnings")
@JsonIdentityInfo(generator = ObjectIdGenerators.PropertyGenerator.class, property = "id")
public class Warning {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    @ManyToOne
    @JoinColumn(name = "member_id", nullable = false)
    private Member member;

    @ManyToOne
    @JoinColumn(name = "warned_by_id", nullable = false)
    private Member warnedBy;

    @Column(nullable = false)
    private String reason;

    @Column(nullable = false)
    private String message;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();
}

File: FrontEnd/cit-forums-ui/src/models/Warning.ts
export interface Warning {
  id: number;
  memberId: number;
  memberUsername: string;
  warnedById: number;
  warnedByUsername: string;
  reason: string;
  message: string;
  createdAt: string;
}

export interface CreateWarningRequest {
  memberId: number;
  reason: string;
  message: string;
}
